app/views/admin/enterprises/form/_primary_details.html.haml

```diff
@@ -42,11 +42,14 @@
     %div{'ofn-with-tip' => t('.visible_in_search_tip')}
       %a= t('admin.whats_this')
   .two.columns
-    = f.radio_button :visible, true
-    = f.label :visible, t('.visible'), value: 'true'
-  .five.columns.omega
-    = f.radio_button :visible, false
-    = f.label :visible, t('.not_visible'), value: 'false'
+    = f.radio_button :visible, "public", 'ng-model' => 'Enterprise.visible'
+    = f.label :visible, t('.visible'), value: 'public'
+  .two.columns
+    = f.radio_button :visible, "hidden", 'ng-model' => 'Enterprise.visible'
+    = f.label :visible, t('.not_visible'), value: 'hidden'
+  .four.columns.omega
+    = f.radio_button :visible, "invisible", 'ng-model' => 'Enterprise.visible'
+    = f.label :visible, t('.invisible'), value: 'invisible'
```

Comment:
I find it difficult to know what the difference between *hidden* and *invisible* is. I can see it from the context but it's confusing. For example, you describe invisible as "Completely hidden".

Naming is hard, I thought of public, private and invisible but maybe we should be more explicit. What do you think about this?

* public - Yes, as public as it gets.
* only_through_links - This is longer but we shouldn't need to write it in many places.
* hidden - It's not completely invisible because you can visit the shop if you know the URL.

Reply:
Sure dont mind going with that. I agree hidden and invisible might hard to distinguish from each other 👍